apps/backend/src/lib/ai/provider-registry.ts

```diff
@@ -0,0 +1,148 @@
+    provider: string
+    modelId: string
+  } {
+    const colonIndex = providerModelString.indexOf(":")
+    if (colonIndex === -1) {
+      throw new Error(
+        `Invalid provider:model format: "${providerModelString}". Expected format: "provider:model_id"`,
+      )
+    }
+
+    const provider = providerModelString.slice(0, colonIndex)
+    const modelId = providerModelString.slice(colonIndex + 1)
+
+    if (!provider || !modelId) {
+      throw new Error(
+        `Invalid provider:model format: "${providerModelString}". Both provider and model_id are required.`,
+      )
+    }
+
+    return { provider, modelId }
+  }
+
+  /**
+   * Check if a provider is supported.
+   */
+  isSupportedProvider(provider: string): provider is SupportedProvider {
+    return SUPPORTED_PROVIDERS.includes(provider as SupportedProvider)
+  }
+
+  /**
+   * Get an AI SDK language model from a provider:model string.
+   * Used by stream-naming-service for generateText().
+   */
+  getModel(providerModelString: string): LanguageModel {
+    const { provider, modelId } = this.parseProviderModel(providerModelString)
+
+    if (!this.isSupportedProvider(provider)) {
+      throw new Error(
+        `Unsupported provider: "${provider}". Supported providers: ${SUPPORTED_PROVIDERS.join(", ")}`,
+      )
+    }
+
+    switch (provider) {
+      case "openrouter":
+        return this.getOpenRouterModel(modelId)
+      default:
+        throw new Error(`Provider "${provider}" is not implemented`)
+    }
+  }
+
+  /**
+   * Get a LangChain ChatModel from a provider:model string.
+   * Used by companion agent for LangGraph.
+   */
+  getLangChainModel(providerModelString: string): ChatOpenAI {
+    const { provider, modelId } = this.parseProviderModel(providerModelString)
+
+    if (!this.isSupportedProvider(provider)) {
+      throw new Error(
+        `Unsupported provider: "${provider}". Supported providers: ${SUPPORTED_PROVIDERS.join(", ")}`,
+      )
+    }
+
+    switch (provider) {
+      case "openrouter":
+        return this.getLangChainOpenRouterModel(modelId)
+      default:
+        throw new Error(`Provider "${provider}" is not implemented for LangChain`)
+    }
+  }
+
+  private getOpenRouterModel(modelId: string): LanguageModel {
+    if (!this.openRouterClient) {
+      throw new Error(
+        "OpenRouter is not configured. Set OPENROUTER_API_KEY environment variable.",
+      )
+    }
+
+    logger.debug({ provider: "openrouter", modelId }, "Creating AI SDK model instance")
+    return this.openRouterClient.chat(modelId)
+  }
+
+  private getLangChainOpenRouterModel(modelId: string): ChatOpenAI {
+    if (!this.openRouterApiKey) {
+      throw new Error(
+        "OpenRouter is not configured. Set OPENROUTER_API_KEY environment variable.",
+      )
+    }
+
+    logger.debug({ provider: "openrouter", modelId }, "Creating LangChain model instance")
+    return new ChatOpenAI({
+      modelName: modelId,
+      openAIApiKey: this.openRouterApiKey,
```

Comment:
Why is this handled differently than `this.openRouterClient`? Could we do this on init the same as with `openRouterClient` instead?